ergotree-interpreter/src/eval/sheader.rs

```diff
@@ -0,0 +1,350 @@
+        assert_eq!(expected as i64, actual);
+    }
+
+    #[test]
+    fn test_eval_height() {
+        let expr = create_get_header_property_expr(sheader::HEIGHT_PROPERTY.clone());
+        let ctx = Rc::new(force_any_val::<Context>());
+        let expected = ctx.headers[HEADER_INDEX].height;
+        let actual = eval_out::<i32>(&expr, ctx.clone());
+        assert_eq!(expected as i32, actual);
+    }
+
+    #[test]
+    fn test_eval_pks() {
+        let ctx = Rc::new(force_any_val::<Context>());
+        let expected = ctx
+            .headers
+            .get(HEADER_INDEX)
+            .map(|h| [h.miner_pk.clone(), h.pow_onetime_pk.clone()])
+            .expect("internal error: empty headers array");
+        let actual = eval_header_pks(ctx.clone());
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn test_eval_pow_distance() {
+        let expr = create_get_header_property_expr(sheader::POW_DISTANCE_PROPERTY.clone());
+        let ctx = Rc::new(force_any_val::<Context>());
+        let expected = ctx.headers[HEADER_INDEX].pow_distance.clone();
+        let actual = {
+            let bi = eval_out::<BigInt256>(&expr, ctx.clone());
+            bi.into()
+        };
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn test_eval_pow_nonce() {
+        let expr = create_get_header_property_expr(sheader::POW_NONCE_PROPERTY.clone());
+        let ctx = Rc::new(force_any_val::<Context>());
+        let expected = ctx.headers[HEADER_INDEX].nonce.clone();
+        let actual = eval_out::<Vec<i8>>(&expr, ctx.clone()).as_vec_u8();
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn test_eval_votes() {
+        let expr = create_get_header_property_expr(sheader::VOTES_PROPERTY.clone());
+        let ctx = Rc::new(force_any_val::<Context>());
+        let expected = ctx.headers[HEADER_INDEX].votes.clone();
+        let actual = {
+            let votes_bytes = eval_out::<Vec<i8>>(&expr, ctx.clone()).as_vec_u8();
+            Votes::try_from(votes_bytes)
+                .expect("internal error: votes bytes buffer length isn't equal to 3")
+        };
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn test_eval_failed_invalid_obj() {
+        // calling for Header property on Context obj
+        let expr: Expr = PropertyCall {
+            obj: Box::new(Expr::Context),
+            method: sheader::VERSION_PROPERTY.clone(),
+        }
+        .into();
+        assert_eq!(
+            try_eval_out_wo_ctx::<i8>(&expr),
+            Err(EvalError::TryExtractFrom(TryExtractFromError(
+                "expected Header, found Context".to_string()
+            )))
+        )
+    }
+
+    #[test]
+    fn test_eval_failed_unknown_property() {
+        let expr = create_get_header_property_expr(sheader::UNKNOWN_PROPERTY.clone());
+        assert_eq!(
+            try_eval_out_wo_ctx::<i8>(&expr),
+            Err(EvalError::NotFound(format!(
```

Comment:
I'd love to catch the exact error in such cases myself. However, this makes tests too fragile. Think of changing the error message wording braking the test. 
I just check if its `is_err()` until a better (non-fragile) solution is found.